### ISSUE TITLE
Add sourceMapPathOverrides

### ISSUE THREAD

**Issue by NatoBoram** [firefox-devtools/vscode-firefox-debug#140]
There's a great starting point for debugging Angular apps at [vscode-recipes/Angular-CLI](https://github.com/Microsoft/vscode-recipes/tree/master/Angular-CLI), but it uses the `sourceMapPathOverrides` key, which doesn't seem to be supported by Debugger for Firefox. Workarounds would be to [add a configuration for Firefox in vscode-recipes/Angular-CLI](https://github.com/microsoft/vscode-recipes/issues/235) or to [add a snippet for Firefox to Angular Snippets](https://github.com/johnpapa/vscode-angular-snippets/issues/99), but it would be nice if Debugger for Firefox could reach feature parity with Debugger for Chrome.

**Comment by hbenl** [comment]
The equivalent to `sourceMapPathOverrides` is `pathMappings`. The syntax is different - instead of `"something/*": "something-else/*"` you have to write `{"url": "something", "path": "something-else"}`. Also, I'm not sure if these are necessary at all - the debug adapter uses some default pathMappings that should work with most webpack setups. I will investigate this soon and report back here.

**Comment by NatoBoram** [comment]
Is there a way to automatically parse `sourceMapPathOverrides` and convert it to `pathMappings`? That would be a nice way to add seamless support for Debugger for Firefox from an existing Debugger for Chrome configuration.

**Comment by hbenl** [comment]
That should be possible, just remove any trailing `*` from the `sourceMapPathOverrides` and convert the way that it is written (i.e. turn `{ "u1": "p1", "u2": "p2" }` into `[{ "url": "u1", "path: "p2" },{ "url": "u2", "path": "p2" }]`).

**Comment by digitarald** [comment]
@hbenl should we make it a feature request to allow the same property and syntax?

**Comment by hbenl** [comment]
@NatoBoram I just created an Angular app with Angular CLI version 6 and I could debug it _without_ any `pathMappings`. So _don't_ try to convert the `sourceMapPathOverrides` from that recipe - I tried that and then breakpoints stopped working.

@digitarald I'm not very fond of that idea - we'd have to ensure that the path mapping code works _exactly_ the same as the one in the Chrome adapter, otherwise any difference will be perceived as a bug in the Firefox adapter (which would only be fair if we create the expectation that it works the same). In my experience, manual path mappings are very error-prone and I've seen several examples where people _break_ the debugger by adding manual path mappings when it would have worked fine without them, so I don't want to encourage people to blindly add even more manual path mappings.
